File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int prevPrice = Integer.MAX_VALUE;
        int profit = 0;
        int possibleProfit = 0;
        
        for(int i = 0; i < prices.length; i++){
            int boughtPrice = prices[i];
            if(boughtPrice < prevPrice){
                prevPrice = boughtPrice;
            }
            possibleProfit = boughtPrice - prevPrice;
            if(profit < possibleProfit){
                profit = possibleProfit;
            }
        }
        return profit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left, right = 0,0
        maxProfit = 0
        while(right != len(prices)):
            leftPrice = prices[left]
            rightPrice = prices[right]
            if(leftPrice < rightPrice):
                profit = rightPrice - leftPrice 
                maxProfit = max(profit, maxProfit)
            else:
                left = right
            right += 1
        return maxProfit
        